solr/core/src/java/org/apache/solr/index/SlowCompositeReaderWrapper.java

```diff
@@ -111,7 +111,12 @@ public static LeafReader wrap(IndexReader reader) throws IOException {
       }
       int createdVersionMajor =
           reader.leaves().get(0).reader().getMetaData().getCreatedVersionMajor();
-      metaData = new LeafMetaData(createdVersionMajor, minVersion, null);
+      metaData =
+          new LeafMetaData(
+              createdVersionMajor,
+              minVersion,
+              null,
+              reader.leaves().get(0).reader().getMetaData().hasBlocks());
```

Comment:
minor/subjective: `reader.leaves().get(0).reader().getMetaData()` assignment to a local variable so that both the `createdVersionMajor` compute above and the `hasBlock` determination here can use it